src/Akeneo/UserManagement/Bundle/Provider/CustomDaoAuthenticationProvider.php

```diff
@@ -0,0 +1,100 @@
+use Akeneo\UserManagement\Component\Model\User;
+use Akeneo\UserManagement\Component\Model\UserInterface;
+use Symfony\Component\Security\Core\Authentication\Provider\DaoAuthenticationProvider;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface as SecurityUserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+
+class CustomDaoAuthenticationProvider extends DaoAuthenticationProvider
+{
+    /** @var UserManager */
+    private $userManager;
+
+    /** @var int */
+    private $accountLockDuration;
+
+    /** @var int */
+    private $accountMaxConsecutiveFailure;
+
+    public function __construct(UserProviderInterface $userProvider, UserCheckerInterface $userChecker, string $providerKey, EncoderFactoryInterface $encoderFactory, UserManager $userManager, bool $hideUserNotFoundExceptions = true, int $accountLockDuration, int $accountMaxConsecutiveFailure)
+    {
+        parent::__construct($userProvider, $userChecker, $providerKey, $encoderFactory, $hideUserNotFoundExceptions);
+        $this->userManager = $userManager;
+        $this->accountLockDuration = $accountLockDuration;
+        $this->accountMaxConsecutiveFailure = $accountMaxConsecutiveFailure;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkAuthentication(\Symfony\Component\Security\Core\User\UserInterface $user, UsernamePasswordToken $token)
+    {
+        assert($user instanceof UserInterface);
+        $this->validateAccountUnlocked($user);
+        try {
+            parent::checkAuthentication($user, $token);
+            $this->resetLockingState($user);
+        } catch (BadCredentialsException $e) {
+            $this->incrementFailureCounter($user);
+            throw $e;
+        }
+    }
+
+    private function incrementFailureCounter(UserInterface $user)
+    {
+        if (is_null($user->getAuthenticationFailureResetDate())) {
+            $user->setAuthenticationFailureResetDate(new \DateTime());
+        }
+        $user->setConsecutiveAuthenticationFailureCounter(
+            1 + $user->getConsecutiveAuthenticationFailureCounter()
+        );
+        $this->userManager->updateUser($user);
+    }
+
+    private function validateAccountUnlocked(UserInterface $user)
+    {
+        if ($this->isCounterReset($user)) {
+            return;
+        }
+        if ($this->isWithinLockTimePeriod($user)
+            && ($user->getConsecutiveAuthenticationFailureCounter() >= $this->accountMaxConsecutiveFailure)) {
+            throw new LockedAccountException($this->accountLockDuration);
+        }
+
+        if (!$this->isWithinLockTimePeriod($user)) {
```

Comment:
```suggestion
        if (!$this->isWithinLockTimePeriod($user)) {
```

I think it should be outside of this method.

Reply:
Only 2 internal occurences, Rule of three can not apply.  :wink: 

...and I would not know where to put such method. it might be interested to introduce a time interval concept...

Reply:
It's not for DRY. A `validate` should not perform any IO or it should be explicit in my opinion.

Reply:
Sorry! 
Yes you're right!  I moved it to the main method...
You now  what i totally missed the point.
Yesterday at late I was thinking of your IO comment. I was thinking of another issue you did not mention: 
\Akeneo\UserManagement\Bundle\Provider\CustomDaoAuthenticationProvider::isWithinLockTimePeriod brings an ugly  side effect with new date... I thought your were mentioning that... :-)